BreadPlayer.Views.UWP/Core/SharedLogic.cs

```diff
@@ -200,85 +200,78 @@ private static (bool NotExists, string FileName) AlbumArtFileExists(Mediafile fi
-                                    using (Windows.Storage.Streams.IRandomAccessStream albumstream = await albumart.OpenAsync(FileAccessMode.ReadWrite))
-                                    {
-                                        while ((buf = (await thumbnail.ReadAsync(inputBuffer, inputBuffer.Capacity, Windows.Storage.Streams.InputStreamOptions.None))).Length > 0)
-                                            await albumstream.WriteAsync(buf);
-                                        return true;
-                                    }
+                                while ((buf = (await thumbnail.ReadAsync(inputBuffer, inputBuffer.Capacity, Windows.Storage.Streams.InputStreamOptions.None))).Length > 0)
+                                    await albumstream.WriteAsync(buf);
+                                return true;
+                            }
 
-                                case ThumbnailType.Icon:
-                                    using (TagLib.File tagFile = TagLib.File.Create(new SimpleFileAbstraction(file), TagLib.ReadStyle.Average))
-                                    {
-                                        if (tagFile.Tag.Pictures.Length >= 1)
-                                        {
-                                            var image = await ApplicationData.Current.LocalFolder.CreateFileAsync(@"AlbumArts\" + albumArt.FileName + ".jpg", CreationCollisionOption.FailIfExists);
+                        case ThumbnailType.Icon:
+                            using (TagLib.File tagFile = TagLib.File.Create(new SimpleFileAbstraction(file), TagLib.ReadStyle.Average))
+                            {
+                                if (tagFile.Tag.Pictures.Length >= 1)
+                                {
+                                    var image = await ApplicationData.Current.LocalFolder.CreateFileAsync(@"AlbumArts\" + albumArt.FileName + ".jpg", CreationCollisionOption.FailIfExists);
 
-                                            using (var albumstream = await image.OpenStreamForWriteAsync())
-                                            {
-                                                await albumstream.WriteAsync(tagFile.Tag.Pictures[0].Data.Data, 0, tagFile.Tag.Pictures[0].Data.Data.Length);
-                                            }
-                                            return true;
-                                        }
+                                    using (var albumstream = await image.OpenStreamForWriteAsync())
+                                    {
+                                        await albumstream.WriteAsync(tagFile.Tag.Pictures[0].Data.Data, 0, tagFile.Tag.Pictures[0].Data.Data.Length);
                                     }
-                                    break;
+                                    return true;
+                                }
                             }
-                            GC.Collect();
```

Comment:
We need this here because GC doesn't collect the closed streams. We have to do it manually so as not to add up extra memory.